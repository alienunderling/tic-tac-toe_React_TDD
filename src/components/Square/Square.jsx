import React from "react";
import PropTypes from "prop-types";
import "./Square.css";

const Square = ({ value, squareAction }) => (
  <button className="square" onClick={squareAction}>
    {value}
  </button>
);

Square.propTypes = {
  value: PropTypes.string.isRequired,
  squareAction: PropTypes.func.isRequired,
};

export default Square;
