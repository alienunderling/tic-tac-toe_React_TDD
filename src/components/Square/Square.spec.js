import React from "react";
import { shallow } from "enzyme";
import Square from "./Square";

describe("Display", () => {
  let wrapper;

  beforeEach(
    () => (wrapper = shallow(<Square value="X" squareAction={jest.fn()} />))
  );

  it("should render a Square <button />", () => {
    expect(wrapper.find("button").length).toEqual(1);
  });

  it("should render the value of value", () => {
    wrapper.setProps({ value: "test" });
    expect(wrapper.text()).toEqual("test");
  });
});
