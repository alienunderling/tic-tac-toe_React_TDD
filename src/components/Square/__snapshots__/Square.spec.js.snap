// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render correctly 1`] = `ShallowWrapper {}`;
